aea/aea_builder.py

```diff
@@ -0,0 +1,746 @@
+            }
+            non_supported_connections = connection_ids_set.difference(
+                all_supported_connection_ids
+            )
+            if len(non_supported_connections) > 0:
+                raise ValueError(
+                    "Connection ids {} not declared in the configuration file.".format(
+                        sorted(map(str, non_supported_connections))
+                    )
+                )
+            connections = [
+                connection
+                for id_, connection in self._package_dependency_manager.connections.items()
+                if id_.public_id in connection_ids_set
+            ]
+        else:
+            connections = list(self._package_dependency_manager.connections.values())
+
+        return connections
+
+    def build(self, connection_ids: Optional[Collection[PublicId]] = None) -> AEA:
+        """
+        Build the AEA.
+
+        :param connection_ids: select only these connections.
+        :return: the AEA object.
+        """
+        wallet = Wallet(self.private_key_paths)
+        identity = self._build_identity_from_wallet(wallet)
+        connections = self._process_connection_ids(connection_ids)
+        for connection in connections:
+            connection.address = identity.address
+            connection.load()
+        aea = AEA(
+            identity,
+            connections,
+            wallet,
+            LedgerApis(self.ledger_apis_config, self._default_ledger),
+            self._resources,
+            loop=None,
+            timeout=0.0,
+            is_debug=False,
+            is_programmatic=True,
+            max_reactions=20,
+        )
+        self._set_agent_context_to_all_skills(aea.context)
+        return aea
+
+    def _set_agent_context_to_all_skills(self, context: AgentContext):
+        """Set a skill context to all skills"""
+        for skill in self._resources.get_all_skills():
+            logger_name = "aea.{}.skills.{}.{}".format(
+                context.agent_name, skill.configuration.author, skill.configuration.name
+            )
+            skill.skill_context.set_agent_context(context)
+            skill.skill_context._logger = logging.getLogger(logger_name)
+
+    def _check_configuration_not_already_added(self, configuration):
+        if (
+            configuration.component_id
+            in self._package_dependency_manager.all_dependencies
+        ):
+            raise ValueError(
+                "Component {} of type {} already added.".format(
+                    configuration.public_id, configuration.component_type
+                )
+            )
+
+    def _check_package_dependencies(self, configuration):
+        self._package_dependency_manager.check_package_dependencies(configuration)
+
+    @staticmethod
+    def _find_component_directory_from_component_id(
+        aea_project_directory: Path, component_id: ComponentId
+    ):
```

Comment:
this is useful to find the component directory from the component id, either vendor or not